### PR TITLE
Add federated_org oidc claim and configure SaaS app claims

### PR DESCRIPTION
**Pull request by sadilchamishka**

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/internal/OrgApplicationMgtDataHolder.java [hunk 1]
```diff
@@ -121,4 +123,12 @@ public void setIdpManager(IdpManager idpManager) {
 
         this.idpManager = idpManager;
     }
+
+    public ClaimMetadataManagementService getClaimMetadataManagementService() {
+        return claimMetadataManagementService;
+    }
+
+    public void setClaimMetadataManagementService(ClaimMetadataManagementService claimMetadataManagementService) {
+        this.claimMetadataManagementService = claimMetadataManagementService;
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/internal/OrgApplicationMgtDataHolder.java:132]: missing new line after the method signature, check other places as well

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/constant/OrgApplicationMgtConstants.java [hunk 2]
```diff
@@ -34,4 +34,10 @@ public class OrgApplicationMgtConstants {
     public static final String UPDATE_SP_METADATA_SHARE_WITH_ALL_CHILDREN = "updateShareWithAllChildren";
     public static final String DELETE_SHARE_FOR_MAIN_APPLICATION = "deleteShareForMainApp";
 
+    public static final String FEDERATED_ORG_CLAIM_URL =  "http://wso2.org/claims/runtime/federated_org";
+
+    public static final String FEDERATED_ORG_CLAIM_DISPLAY_NAME = "federated_org";
+
+    public static final String OIDC_CLAIM_DIALECT_URI = "http://wso2.org/oidc/claim";
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/constant/OrgApplicationMgtConstants.java:41]: don't level newlines in between. as these constant set are related

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/internal/OrgApplicationMgtDataHolder.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/internal/OrgApplicationMgtDataHolder.java:134]: Fixed by - https://github.com/wso2-extensions/identity-organization-management/pull/166/commits/6641182599629c63da7fd052745660558c848adf

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/constant/OrgApplicationMgtConstants.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/constant/OrgApplicationMgtConstants.java]: Fixed by - https://github.com/wso2-extensions/identity-organization-management/pull/166/commits/6641182599629c63da7fd052745660558c848adf

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 3]
```diff
@@ -174,6 +191,52 @@ public void shareOrganizationApplication(String ownerOrgId, String originalAppId
...
+                    .getExternalClaims(OIDC_CLAIM_DIALECT_URI, tenantDomain).stream()
+                    .filter(externalClaim -> FEDERATED_ORG_CLAIM_DISPLAY_NAME.equals(externalClaim.getClaimURI()))
+                    .findAny();
+            if (!optionalExternalClaim.isPresent()) {
+                getClaimMetadataManagementService().addExternalClaim(new ExternalClaim(OIDC_CLAIM_DIALECT_URI,
+                        FEDERATED_ORG_CLAIM_DISPLAY_NAME, FEDERATED_ORG_CLAIM_URL), tenantDomain);
+            }
+
+            ScopeDTO scopeDTO = getOAuthAdminService().getScope(OPENID);
+            Optional<String> optionalOidcClaim = stream(scopeDTO.getClaim())
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java:220]:
```suggestion
            Optional<String> optionalOIDCClaim = stream(scopeDTO.getClaim())
```

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java (hunk 3 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Fixed by - https://github.com/wso2-extensions/identity-organization-management/pull/166/commits/b324a1504385d0cf11ba431b4948c05d11c48974

**Pull request merged**